test/test_public_bindings.py

```diff
@@ -261,7 +248,14 @@ def test_correct_module_names(self):
 
         def test_module(modname):
             split_strs = modname.split('.')
-            mod = sys.modules.get(modname)
+            try:
```

Comment:
To check my understanding: let's say someone added a new torch.foo.bar module, but did not modify allowlist_for_publicAPI.json. Will our testing catch it?

Reply:
If their module introduces new public APIs that are not compliant, this will fail yes.
The different is that before, we only looked into `sys.modules` which are the modules already imported. Now we import the module itself so we can check everything.

Reply:
sgtm, that's what I was looking for